xfer/publisher.go

```diff
@@ -52,16 +53,24 @@ func NewHTTPPublisher(target, token, id string) (*HTTPPublisher, error) {
 
 // Publish publishes the report to the URL.
 func (p HTTPPublisher) Publish(rpt report.Report) error {
-	var buf bytes.Buffer
-	if err := gob.NewEncoder(&buf).Encode(rpt); err != nil {
+	gzbuf := new(bytes.Buffer)
+	gzwriter := gzip.NewWriter(gzbuf)
```

Comment:
I tried that, but there was a problem with data not being flushed. I guess someone is missing a `Close()` call...